astropy/coordinates/representation.py

```diff
@@ -324,6 +324,17 @@ def __sub__(self, other):
     def __rsub__(self, other):
         return self._combine_operation(operator.sub, other, reverse=True)
 
+    # Required to check equality between representation and differential
+    # classes
+    def __eq__(self, other):
+        if self.__class__ != other.__class__:
+            return False
+
+        for component in self.components:
+            if getattr(self, component) != getattr(other, component):
```

Comment:
This will not work for non-scalar components, since then you get an array of bool